packages/manager/modules/anthos/src/dashboard/components/order-public-ip/controller.js

```diff
@@ -1,7 +1,68 @@
+import { MAX_QUANTITY, PRODUCT_ID, PRICING_DURATION } from './constants';
+import { extractPublicIpsAddonFromAnthosCatalog } from './utils';
+
 export default class {
-  onOrderPublicCancel() {
-    this.trackClick(`${this.orderPublicIpHitTracking}::cancel`);
+  /* @ngInject */
+  constructor($q, $translate, AnthosTenantsService, User) {
+    this.$q = $q;
+    this.$translate = $translate;
+    this.AnthosTenantsService = AnthosTenantsService;
+    this.User = User;
+
+    this.addon = null;
+    this.expressOrderUrl = '';
+    this.quantity = 1;
+    this.step = 1;
+    this.isLoading = true;
+    this.MAX_QUANTITY = MAX_QUANTITY;
+  }
+
+  $onInit() {
+    this.$q
+      .all({
+        anthosCatalog: this.AnthosTenantsService.getAnthosCatalog(),
+        expressOrderUrl: this.User.getUrlOf('express_order'),
+      })
+      .then(({ anthosCatalog, expressOrderUrl }) => {
+        this.addon = extractPublicIpsAddonFromAnthosCatalog(anthosCatalog);
+        this.expressOrderUrl = expressOrderUrl;
+        if (!this.addon) throw new Error('missingAddon');
```

Comment:
Shouldn't this be `displayAlerterMessage` as well?

Reply:
Yes it is in the chained catch handler